Control/Monad/Logic.hs

```diff
@@ -176,11 +196,14 @@ logic f = LogicT $ \k -> Identity .
 -- >>> observe empty
 -- *** Exception: No answer.
 --
+-- Since 'Logic' is isomorphic to a list, 'observe' is 'head' in disguise.
+--
 observe :: Logic a -> a
 observe lt = runIdentity $ unLogicT lt (const . pure) (error "No answer.")
 
 -------------------------------------------------------------------------
--- | Extracts all results from a 'Logic' computation.
+-- | Extracts all results from a 'Logic' computation, witnessing
+-- a half of isomorphism between 'Logic' and lists.
 --
 -- >>> observe (pure 5 <|> empty <|> empty <|> pure 3 <|> empty)
```

Comment:
Oops!  Not something exactly pertinent to this PR, but it looks like this should have been `observeAll` in this example.